proxy/logging/LogUtils.cc

```diff
@@ -478,6 +478,41 @@ LogUtils::seconds_to_next_roll(time_t time_now, int rolling_offset, int rolling_
   return ((tr >= sidl ? (tr - sidl) % rolling_interval : (86400 - (sidl - tr)) % rolling_interval));
 }
 
+ts::TextView
+LogUtils::get_unrolled_filename(ts::TextView rolled_filename)
+{
+  auto unrolled_name = rolled_filename;
+
+  // A rolled log will look something like:
+  //   squid.log_some.hostname.com.20191029.18h15m02s-20191029.18h30m02s.old
+  auto suffix = rolled_filename;
+
+  suffix.remove_prefix_at('.');
+  // Using the above squid.log example, suffix now looks like:
+  //   log_some.hostname.com.20191029.18h15m02s-20191029.18h30m02s.ol
+
+  if (suffix.find('_') != std::string::npos) {
```

Comment:
You might look at `split_prefix_at` here, which does basically this. E.g.
```
if (suffix.split_prefix_at('_') || suffix.split_prefix_at('.')) {
  return unrolled_name.remove_suffix(suffix.size() + 1);
}
return rolled_filename;
```

Reply:
Cool.